### PR TITLE
Make the app build again

### PR DESCRIPTION
**Pull request by fdrouet**
- add Fastlane for Appaloosa and Google Play store publishing
- configure the APK signing for release build
- target sdk version is now 26 (v27 doesn't compile yet)
- upgrade Gradle to 4.1
- improve the gradle build configurability
  - add product Flavor for store and beta (Appaloosa)
  - add version and application Id suffix if the build target is not the Google Play Store

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,22 @@
-[![Build Status](https://travis-ci.org/exoplatform/exo-android.svg)](https://travis-ci.org/exoplatform/exo-android)
+# ATIS Network Android application
```
Comment (thomasdelhomenie) [inline README.md]: To remove.

File: README.md [hunk 2]
```diff
@@ -1,3 +1,22 @@
...
 
-# exo-hybrid-android
+This project is based on a fork of [eXo Mobile Android application](https://github.com/exoplatform/exo-android/)
```
Comment (thomasdelhomenie) [inline README.md]: To remove.

File: app/build.gradle [hunk 3]
```diff
@@ -1,59 +1,89 @@
...
     defaultConfig {
         applicationId "org.exoplatform"
-        minSdkVersion 16
-        targetSdkVersion 27
-        versionCode 1604281805
-        versionName "5.0"
+        minSdkVersion rootProject.ext.minSdkVersion
+        targetSdkVersion rootProject.ext.compileSdkVersion
+        versionCode 27
+        versionName "5.0.0-SNAPSHOT"
```
Comment (thomasdelhomenie) [inline app/build.gradle]: We use 5.0.x-SNAPSHOT for snapshots.

File: build.gradle [hunk 4]
```diff
@@ -1,24 +1,47 @@
...
+        }
+    }
     repositories {
         maven { url "https://repository.exoplatform.org/public/" }
         mavenLocal()
     }
+
+    afterEvaluate { project ->
+        // Load signing configuration
+        def propsFile = file(System.getenv('GOOGLE_PLAY_EXO_KEYSTORE_PROPS'))
```
Comment (thomasdelhomenie) [inline build.gradle]: I have this error if I didn't set this variable (case of developers) :

> Error:(23, 1) A problem occurred configuring root project 'exo-android'.
path may not be null or empty string. path='null'

We should be able to launch the application in dev mode (emulator) without tweaking the build file IMO.

File: app/build.gradle [hunk 5]
```diff
@@ -1,59 +1,89 @@
...
+    signingConfigs {
+        release
     }
     useLibrary 'org.apache.http.legacy'
+
     buildTypes {
         release {
             minifyEnabled false
+            useProguard false
+            signingConfig signingConfigs.release
```
Comment (thomasdelhomenie) [inline app/build.gradle:33]: I have this error :

> Error:Execution failed for task ':app:validateSigningBetaRelease'.
Keystore file not set for signing config release

File: build.gradle [hunk 6]
```diff
@@ -1,24 +1,47 @@
...
+            android.signingConfigs[configName].keyPassword = props['keyPassword']
+        }
+    }
 }
 
 task clean(type: Delete) {
     delete rootProject.buildDir
 }
+
+// This block encapsulate custom properties and make theme available to all modules in the project.
```
Comment (thomasdelhomenie) [inline build.gradle]: "theme" ?

**Review by thomasdelhomenie — CHANGES_REQUESTED** [review]

File: app/build.gradle (hunk 5 above)
Comment (fdrouet) [reply app/build.gradle:33]: which command do you type to have this error ?

File: build.gradle (hunk 6 above)
Comment (fdrouet) [reply build.gradle]: > them ;-)

File: app/build.gradle (hunk 5 above)
Comment (thomasdelhomenie) [reply app/build.gradle:33]: I just try to run the app in Android Studio (launch the emulator).

File: build.gradle (hunk 6 above)
Comment (thomasdelhomenie) [reply build.gradle]: ok :smile:

File: app/build.gradle (hunk 5 above)
Comment (fdrouet) [reply app/build.gradle:33]: You should not use the betaRelease variant but the localDebug one.

**Pull request merged**